core/src/main/java/com/scalar/db/api/Scan.java

```diff
@@ -457,4 +484,316 @@ public enum Order {
       DESC,
     }
   }
+
+  @Immutable
+  public static class Conjunction {
+    private final Set<ConditionalExpression> conditions;
+
+    private Conjunction(Set<ConditionalExpression> conditions) {
+      this.conditions = conditions;
+    }
+
+    /**
+     * Returns the set of {@code ConditionalExpression} which this conjunction is composed of.
+     *
+     * @return set of {@code ConditionalExpression} which this conjunction is composed of
+     */
+    public Set<ConditionalExpression> getConditions() {
+      return ImmutableSet.copyOf(conditions);
+    }
+
+    /**
+     * Creates a {@code Conjunction} object with a single conditional expression.
+     *
+     * @param condition a conditional expression
+     * @return a {@code Conjunction} object
+     */
+    public static Conjunction of(ConditionalExpression condition) {
+      return new Conjunction(ImmutableSet.of(condition));
+    }
+
+    /**
+     * Creates a {@code Conjunction} object with a collection of conditional expressions.
+     *
+     * @param conditions a collection of conditional expressions
+     * @return a {@code Conjunction} object
+     */
+    public static Conjunction of(Collection<ConditionalExpression> conditions) {
+      return new Conjunction(ImmutableSet.copyOf(conditions));
+    }
+
+    /**
+     * Creates a {@code Conjunction} object with conditional expressions.
+     *
+     * @param conditions conditional expressions
+     * @return a {@code Conjunction} object
+     */
+    public static Conjunction of(ConditionalExpression... conditions) {
+      return new Conjunction(ImmutableSet.copyOf(conditions));
+    }
+
+    /**
+     * Indicates whether some other object is "equal to" this object. The other object is considered
+     * equal if:
+     *
+     * <ul>
+     *   <li>it is also an {@code Conjunction}
+     *   <li>both instances have the same set of {@code ConditionalExpression}
+     * </ul>
+     *
+     * @param o an object to be tested for equality
+     * @return {@code true} if the other object is "equal to" this object otherwise {@code false}
+     */
+    @Override
+    public boolean equals(Object o) {
+      if (o == this) {
+        return true;
+      }
+      if (!(o instanceof Conjunction)) {
+        return false;
+      }
+      Conjunction other = (Conjunction) o;
+      return conditions.equals(other.conditions);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(conditions);
+    }
+
+    @Override
+    public String toString() {
+      return conditions.toString();
+    }
+  }
+
+  @Immutable
+  public static class AndConditionSet {
```

Comment:
Sorry, `Conjunction` and `AndConditionSet` sound similar to me. Could you add a comment to clarify what they are ?

Reply:
Good point. Essentially, both are the same. But I intentionally used the different classes for the internal representation (`Conjunction`) and user API (`AndConditionSet`) for ease of understanding since the conjunction would be unfamiliar for many users. From the developer's perspective, I just prefer the term "conjunction" since it's short and technically correct to use many places internal modules, including gRPC proto. Do you think "internal use" or something like that is enough here? Or should we use `AndConditionSet` also for the internal representation?

Reply:
So maybe we should move `AndConditionSet` and `OrConditionSet` and their builder classes into `ScanBuilder`?

Reply:
I see. Thanks for the explanation. Either option, adding a comment that describes the relationship between `Conjunction` and `AndConditionSet` and the usages or using `AndConditionSet` for internal use, sounds fine with me.

Reply:
@komamitsu @brfrn169 Thanks for the comments. I will move both sets to the builder and describe the relationship and usage.